packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -273,17 +280,21 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
 
     this.loadBalancer = props.loadBalancer !== undefined ? props.loadBalancer : new ApplicationLoadBalancer(this, 'LB', lbProps);
 
-    const targetProps = {
-      port: 80
```

Comment:
What you could do is leave this as is, and just replace `port: 80` with `port: props.listenerPort !== undefined ? props.listenerPort : 80;`

The rest of the changes below are not necessary to support listener ports.

Reply:
Thanks for the insight. I will continue work on it tomorrow. Great to have learned a lot from it.

And will add the integ and unit testing as well.

Reply:
This has yet to be resolved.

Reply:
I have made some fix per requested and all tests passed. Please review it again. Thanks.